arch/riscv/core/thread.c

```diff
@@ -17,7 +17,7 @@
  * Glogal variable used to know the current mode running.
  * Is not boolean because it must match the PMP granularity of the arch.
  */
-ulong_t is_user_mode;
+uint32_t is_user_mode;
 bool irq_flag;
 #endif
 
```

Comment:
Why do you change this type?
This is_user_mode is used at arch/riscv/core/isr.S and accessed by "sb" instruction (= byte access).
It seems uint32_t and ulong_t are both OK.

Reply:
Thanks for reviewing!

Because `is_user_mode` is protected by 4-byte PMP region in `z_riscv_init_user_accesses()`, `is_user_mode` will be 8-byte and be larger than PMP region in RV64.
```
struct riscv_pmp_region dynamic_regions[] = {
  {
    /* MCU state */
    .start = (ulong_t) &is_user_mode,
    .size = 4,
    .perm = PMP_R,
    .mode = PMP_MODE_NA4,
  }, ...

// p.s. Origin RISC-V PMP code will use 4-byte PMP region for it, I haven't change it in my PMP patch.
```

`is_user_mode` only needs 1-byte so I don't enlarge PMP region but shrink size of `is_user_mode`  in RV64.

Reply:
Ah, this area is protected by PMP... I understand.
Thanks!